lib/stub_ui/lib/painting.dart

```diff
@@ -1023,6 +1023,46 @@ enum Clip {
 // constant and can't propagate into the set/get calls.
 const Endian _kFakeHostEndian = Endian.little;
 
+/// An immutable image dimensions container that holds the target dimensions.
+class ImageResizeDimensions {
+  /// Creates [ImageResizeDimensions] with given width and height.
+  const ImageResizeDimensions(this.width, this.height);
+
+  /// Creates [ImageResizeDimensions] with the given width, while
+  /// preserving the aspect ratio of the original image.
+  factory ImageResizeDimensions.resizeWidthPreserveAspectRatio(int widthParam) {
+    return ImageResizeDimensions(widthParam, _kDoNotResizeDimension);
+  }
+
+  /// Creates [ImageResizeDimensions] with the given height, while
+  /// preserving the aspect ratio of the original image.
+  factory ImageResizeDimensions.resizeHeightPreserveAspectRatio(int heightParam) {
+    return ImageResizeDimensions(_kDoNotResizeDimension, heightParam);
+  }
+
+  /// Creates [ImageResizeDimensions] which preserves the original dimensions
+  /// of the image.
+  static const ImageResizeDimensions kDoNotResize = ImageResizeDimensions(_kDoNotResizeDimension, _kDoNotResizeDimension);
+
+  static const int _kDoNotResizeDimension = -1;
+
+  /// Horizontal extent of this image.
+  final int width;
+
+  /// Vertical extent of this image.
+  final int height;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other is! ImageResizeDimensions) return false;
+    final ImageResizeDimensions typedOther = other;
+    return width == typedOther.width && height == typedOther.height;
+  }
+
+  @override
+  int get hashCode => hashValues(width, height);
+}
+
```

Comment:
Rather than introducing new API surface, we should use `Size`. I recommend using `null` to represent the "no fixed size" edges.

Reply:
Answered in the previous comment. About using `null`, I was hoping that using `-1` would let us reference the same placeholder value in both dart as well as c++ land. Else, we would be using `null` and some other optional placeholder value in c++.

Reply:
In that case I would just have `int width, int height` arguments to `instantiateImageCodec`. Having a whole class and so on is a lot of API surface to add for something we're not going to use widely.